src/Data/List/Lazy.purs

```diff
@@ -705,3 +706,11 @@ foldM f a xs =
          Nothing -> pure a
          Just { head: b, tail: bs } ->
                        f a b >>= \a' -> foldM f a' bs
+
+-- | Perform a right fold lazily
+foldr' :: forall a b. Z.Lazy b => (a -> b -> b) -> b -> List a -> b
```

Comment:
Ah, I thought you knew of a way and were attempting to deliver it from me Socratically :p